### PR TITLE
fix: guard empty OpenAI tool names in Claude translation

### PR DESCRIPTION
**Pull request by chow651**
## Summary

### CODE REVIEW

**Comment by chow651** [comment]
Additional repro context from a real Claude Code setup:

**Comment by chow651** [comment]
Addressed the latest review feedback in follow-up commit `647e19f`:

File: internal/translator/openai/claude/openai_claude_request.go [hunk 1]
```diff
@@ -176,13 +183,21 @@ func ConvertClaudeRequestToOpenAI(modelName string, inputRawJSON []byte, stream
 								toolCallJSON, _ = sjson.SetBytes(toolCallJSON, "function.arguments", "{}")
 							}
 
+							if toolCallID != "" {
+								validToolUseIDs[toolCallID] = struct{}{}
+							}
```
[the remark this answers is not in the document]
Comment (chow651) [reply internal/translator/openai/claude/openai_claude_request.go]: Handled in f217181. Empty Claude tool_use IDs are now synthesized into stable OpenAI tool_call IDs, empty tool_result.tool_use_id values consume those generated IDs in order, and explicit IDs still map directly. Added regression coverage for single-empty, multi-empty, and mixed explicit/generated ID histories.

**Comment by chow651** [comment]
@codex review

**Comment by chow651** [comment]
@codex review

**Comment by chow651** [comment]
@codex review

**Comment by chow651** [comment]
@codex review

**Comment by chow651** [comment]
I ran into these issues while using Claude Code against an OpenAI-compatible relay and ended up patching them locally to get unstuck. After tracing the failures down, I thought the fixes and regression coverage might also be useful here, so I opened this PR. I’m still fairly new to contributing in projects like this, and I also handled the branch targeting wrong at first.

**Comment by chow651** [comment]
@codex review

**Review by luispater — APPROVED** [review]
### Summary
This PR hardens OpenAI↔Claude tool-call translation to avoid emitting invalid tool history:
- Skip tool calls/tool_use blocks when the tool name is empty (and drop orphan tool results) to prevent upstream 400s / polluted Claude history.
- Generate unique OpenAI tool_call IDs when Claude tool_use IDs are empty or duplicated, and scope tool-result queues to the current tool turn.
- Improve stop_reason mapping so `tool_use` is only reported when a valid tool_use block was actually emitted, while preserving `length`→`max_tokens` semantics.

### Findings
No blocking issues found. The edge-case handling is comprehensive and backed by focused unit tests covering streaming and non-streaming paths (empty tool name, delayed tool name, missing/duplicate IDs, stale tool_result queues).

### Test plan
- Local (merge result = latest `dev` + this PR): `go test ./...`

This is an automated Codex review result and still requires manual verification by a human reviewer.

**Comment by luispater** [comment]
Thanks for the PR and the detailed reproduction.

The issue was fixed separately in c1caa454 with a smaller scoped change focused on the streaming response translator, plus a regression test for the `function.name:null` chunk. I am closing this PR to avoid merging the broader request/response/tool-id changes as part of this narrow fix.
